### PR TITLE
Add UTD80 PCB

### PR DESCRIPTION
**Pull request by dominicgan**
Adds the UTD80 pcb to QMK. Following up on the previous PR that seems to have been abandoned by the original owner.

2020/02/09 - Add via keymap/compatibility.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Continuing #7701 (abandoned by owner)

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/utd80/config.h [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#pragma once
```
Comment (drashna) [inline keyboards/utd80/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/utd80/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/utd80/keymaps/default/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/utd80/utd80.c [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+   setPinOutput(B6);
+}
+
+bool led_update_kb(led_t led_state) {
+    if(led_update_user(led_state)) {
+        writePin(B6, led_state.caps_lock);
+        writePin(B5, led_state.scroll_lock);
+    }
+    return true;
+}
```
Comment (drashna) [inline keyboards/utd80/utd80.c:22]: All of this can be replaced with a define. or two. 
```suggestion
```

File: keyboards/utd80/config.h [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+#define BACKLIGHT_PIN B7
+#define BACKLIGHT_BREATHING
+#ifdef BACKLIGHT_PIN
+#define BACKLIGHT_LEVELS 3
+#endif
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* define if matrix has ghost */
```
Comment (drashna) [inline keyboards/utd80/config.h:31]:
```suggestion
#define LED_CAPS_LOCK_PIN B6
#define LED_SCROLL_LOCK_PIN B5

/* define if matrix has ghost */
```

File: keyboards/utd80/utd80.c [hunk 5]
```diff
@@ -0,0 +1,22 @@
+#include "utd80.h"
```
Comment (drashna) [inline keyboards/utd80/utd80.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/utd80/utd80.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
+#pragma once
```
Comment (drashna) [inline keyboards/utd80/utd80.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by spidey3 — COMMENTED** [review]
Just to be clear - I see from your description that you have picked this up when the original maintainer (UTDKeyboard) didn't finish getting their PR approved and merged. But I note that you've added their company name to the copyright notice, and not your own name. Is that what you intend? Are you associated with UTDKeyboard, or have you spoken with them about this?

**Comment by dominicgan** [comment]
@spidey3 Hello, no I'm not affiliated with the company, I was just picking up on the changes requested where the previous PR #7701 was left off. I used their company name since I didn't contribute much aside from cleaning it up and adding a via keymap. 

I have tried to contact them to no avail so I'm working on this independently. If it helps I could change the attribution?

**Comment by spidey3** [comment]
> @spidey3 Hello, no I'm not affiliated with the company, I was just picking up on the changes requested where the previous PR #7701 was left off. I used their company name since I didn't contribute much aside from cleaning it up and adding a via keymap.

I think I would change the attribution to "UTDKeyboard & Dominic Gan", just to be accurate.

**Comment by dominicgan** [comment]
> I think I would change the attribution to "UTDKeyboard & Dominic Gan", just to be accurate.

will do, thanks!

**Review by spidey3 — APPROVED** [review]

**Pull request merged**